internal/controllers/framework_controller.go

```diff
@@ -110,15 +120,16 @@ func (r *FrameworkReconciler) reconcile(ctx context.Context, framework *ketchv1.
 			time.Sleep(1 * time.Second)
 		}
 	}
+
+	cleanAndUpdateMap(&namespace.Annotations, framework.Spec.Annotations)
```

Comment:
I think it is possible to guess that `namespace.Annotations` will be updated, 
can I suggest to show the intention in the following way:

```golang
namespace.Annotations = clean(framework.Spec.Annotations)   // or copy()
```

?